### PR TITLE
langref: document extern variadic functions

### PR DESCRIPTION
**Pull request by perillo**
Add a new subsection within the C section, documenting extern variadic functions.

### CODE REVIEW

File: doc/langref.html.in [hunk 1]
```diff
@@ -10792,6 +10792,30 @@ pub const MAKELOCAL = @compileError("unable to translate C expr: unexpected toke
...
+const testing = std.testing;
+
+pub extern "c" fn printf(format: [*:0]const u8, ...) c_int;
+
+test "variadic function" {
+    try testing.expect(printf("Hello, world!\n") == 14);
+    try testing.expect(@typeInfo(@TypeOf(printf)).Fn.is_var_args);
+}
+      {#code_end#}
+      <p>Non extern variadic functions are not allowed.</p>
```
Comment (Vexu) [inline doc/langref.html.in:10810]: Non-extern variadic functions are an accepted proposal #515

Comment (perillo) [reply doc/langref.html.in:10810]: I was aware of #515; do you think that I should add a link to the issue?

Comment (Vexu) [reply doc/langref.html.in:10810]: It shouldn't say that they are not allowed, linking the issue and saying it hasn't been implemented would be better.

Comment (perillo) [reply doc/langref.html.in:10810]: Thanks.

By the way: is it correct to do a `git commit --amend` followed by `git push  --force-with-lease` ?

Comment (Vexu) [reply doc/langref.html.in]: Yes, but you can also just push more commits if you want and then the PR can be squash merged.

**Pull request merged**